src/main/java/org/embulk/output/hdfs/client/HdfsClient.java

```diff
@@ -0,0 +1,266 @@
+                    throws Exception
+            {
+                return fs.delete(path, recursive);
+            }
+        });
+    }
+
+    public void globAndRemove(final Path globPath)
+    {
+        for (final FileStatus fileStatus : glob(globPath)) {
+            if (fileStatus.isDirectory()) {
+                logger.debug("Skip {} because {} is a directory.",
+                        fileStatus.getPath(), fileStatus.getPath());
+                continue;
+            }
+            logger.debug("Remove: {}", fileStatus.getPath());
+            boolean isRemoved = delete(fileStatus.getPath(), false);
+            if (!isRemoved) {
+                throw new RuntimeException(String.format("Remove Failed: %s", fileStatus.getPath()));
+            }
+        }
+    }
+
+    public void globAndRemoveRecursive(final Path globPath)
+    {
+        for (final FileStatus fileStatus : glob(globPath)) {
+            logger.debug("Remove: {}", fileStatus.getPath());
+            boolean isRemoved = delete(fileStatus.getPath(), true);
+            if (!isRemoved) {
+                throw new RuntimeException(String.format("Remove Failed: %s", fileStatus.getPath()));
+            }
+        }
+    }
+
+    public OutputStream create(final Path path, final boolean overwrite)
+    {
+        return run(new Retryable<OutputStream>()
+        {
+            @Override
+            public OutputStream call()
+                    throws Exception
+            {
+                return fs.create(path, overwrite);
+            }
+        });
+    }
+
+    public boolean mkdirs(String path)
+    {
+        return mkdirs(new Path(path));
+    }
+
+    public boolean mkdirs(final Path path)
+    {
+        return run(new Retryable<Boolean>() {
+            @Override
+            public Boolean call()
+                    throws Exception
+            {
+                return fs.mkdirs(path);
+            }
+        });
+    }
+
+    public void close()
+    {
+        run(new Retryable<Void>() {
+            @Override
+            public Void call()
+                    throws Exception
+            {
+                fs.close();
+                return null;
+            }
+        });
+    }
+
+    public void renameDirectory(String src, String dst, boolean overwrite)
+    {
+        renameDirectory(new Path(src), new Path(dst), overwrite);
+    }
+
+    public Void renameDirectory(final Path src, final Path dst, final boolean overwrite)
+    {
+        return run(new Retryable<Void>() {
+            @Override
+            public Void call()
+                    throws Exception
+            {
+                if (fs.exists(dst)) {
+                    if (!overwrite) {
+                        throw new DataException(String.format("Directory Exists: %s", dst.toString()));
+                    }
+                    logger.info("Overwrite: {} >>> {}", src, dst);
+                    if (!new Trash(conf).moveToTrash(dst)) {
```

Comment:
Precisely, this is not atomic.

```
mv dst trash
mv src dst
```

You may need to note about this on README.

For example, embulk-output-mysql 'mode: replace' is telling as it is not atomic https://github.com/embulk/embulk-output-jdbc/tree/master/embulk-output-mysql#modes.

Reply:
Thanks, I follow the convention 8e8de7e